### PR TITLE
Sonariac-448 Docker: parse STOPSIGNAL instruction

### PR DESCRIPTION
**Pull request by mstachniuk**

### CODE REVIEW

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/TreeFactory.java [hunk 1]
```diff
@@ -48,4 +50,8 @@ public MaintainerTree maintainer(SyntaxToken maintainerToken, List<SyntaxToken>
   public SyntaxToken argument(SyntaxToken token) {
     return token;
   }
+
+  public StopSignalTree stopsignal(SyntaxToken token, SyntaxToken tokenValue) {
```
Comment (nils-werner-sonarsource) [inline iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/TreeFactory.java:54]: Why is the method name in lowercase only and the class name in PascelCase on the keyword?

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/StopSignalTreeImpl.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+import java.util.List;
+import org.sonar.iac.common.api.tree.Tree;
+import org.sonar.iac.docker.tree.api.StopSignalTree;
+import org.sonar.iac.docker.tree.api.SyntaxToken;
+
+public class StopSignalTreeImpl extends DockerTreeImpl implements StopSignalTree {
+
+  private final SyntaxToken stopSignalToken;
+  private final SyntaxToken stopSignalValueToken;
```
Comment (nils-werner-sonarsource) [inline iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/StopSignalTreeImpl.java:30]: Maybe `instructionKeyword` and `signal` are more handy names.

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/grammar/DockerGrammar.java [hunk 3]
```diff
@@ -75,4 +81,13 @@ public List<SyntaxToken> ARGUMENTS() {
       )
     );
   }
+
+  public StopSignalTree STOPSIGNAL() {
+    return b.<StopSignalTree>nonterminal(DockerLexicalGrammar.STOPSIGNAL).is(
+      f.stopsignal(
+        b.token(DockerKeyword.STOPSIGNAL),
+        b.token(STRING_LITERAL)
```
Comment (nils-werner-sonarsource) [inline iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/grammar/DockerGrammar.java:89]: Are multiple signals also allowed?

**Review by nils-werner-sonarsource — APPROVED** [review]
LGTM! Please see my comments

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/grammar/DockerGrammar.java (hunk 3 above)
Comment (rudy-regazzoni-sonarsource) [reply iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/grammar/DockerGrammar.java:89]: I tried, multiple signal is not possible in one `STOPSIGNAL` instruction.

File: iac-extensions/docker/src/test/java/org/sonar/iac/docker/tree/impl/StringLiteralTest.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  @Test
+  void test() {
+    Assertions.assertThat(DockerLexicalGrammar.STRING_LITERAL)
+      .matches("f")
+      .matches("foo")
+      .matches("   foo")
+      .matches("1")
+      .matches("123")
+      .matches("SIGKILL");
+  }
```
Comment (rudy-regazzoni-sonarsource) [inline iac-extensions/docker/src/test/java/org/sonar/iac/docker/tree/impl/StringLiteralTest.java:37]: Good idea for putting test on the `STRING_LITERAL` !
I would add few more use case, especially about `STRING_LITERAL_WITH_QUOTES` versions :
`"\"mystring\""`
`"\"partial_quotes_1"`
`"partial_quotes_2\""`
` ` (empty)

**Review by rudy-regazzoni-sonarsource — APPROVED** [review]
LGTM !

**Pull request merged**